### PR TITLE
add stock handling in order route

### PR DESCRIPTION
**Pull request by DenisMarine**

### CODE REVIEW

File: back-end-g2/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "private": true,
   "scripts": {
     "start": "nodemon ./bin/www.js",
-    "test": "cross-env NODE_ENV=test NODE_OPTIONS='--experimental-vm-modules' jest --forceExit --testTimeout=10000",
+    "test": "cross-env NODE_ENV=test NODE_OPTIONS='--experimental-vm-modules' jest tests/orders.test.js --forceExit --testTimeout=10000",
```
Comment (IncSow) [inline back-end-g2/package.json:8]: ? why

File: back-end-g2/src/routes/orders/orderRoutes.js [hunk 2]
```diff
@@ -151,7 +172,21 @@ const orderRoutes = ({ app }) => {
...
         .returning("*")
 
-      await CartModel.query().findById(cart_id).patch({ is_active: false })
+      await CartModel.query()
+        .findById(cart_id)
+        .patch({ is_active: false })
+        .withGraphFetched("all_products")
+        .returning("*")
+
+      await Promise.all(
```
Comment (IncSow) [inline back-end-g2/src/routes/orders/orderRoutes.js:181]: Careful; Right now this is done when you generate the payment link, not when the person pays!

**Review by IncSow — CHANGES_REQUESTED** [review]

File: back-end-g2/package.json (hunk 1 above)
Comment (DenisMarine) [reply back-end-g2/package.json:8]: oops

File: back-end-g2/src/routes/orders/orderRoutes.js (hunk 2 above)
Comment (DenisMarine) [reply back-end-g2/src/routes/orders/orderRoutes.js:181]: where we now if the person pay ???

Comment (DenisMarine) [reply back-end-g2/src/routes/orders/orderRoutes.js]: find it

**Review by IncSow — APPROVED** [review]

**Pull request merged**